woocommerce/woocommerce-ios
# Edit Product: add and remove tags from a Product

Closes #2081 

## Description
This will be the last part for managing Product Tags. In particular, the focus of this PR is on the UI and on the logic behind it. I've to say that making it work exactly like in the WordPress app was a little bit tricky because our models are a little bit different. In the WordPress app, the tags are managed as simple strings, but in our case, there is a `ProductTag` entity, so I needed to create a method `mergeTags:` for merging tag string entities with the existing/fetched/added `ProductTag` entities.
In this PR I updated only the existing unit tests based on changed that I made, but I've not included any test because the PR is big, and I don't want to make it more complex. I'll take care to test some methods, in particular, the one used for the Tag tokenization and for Text & Input Handling in a separate PR.

## Changes
- There was an error on `ProductStore`. The updates, inserts, or prunes of the provided StorageProduct's tags using the provided read-only Product's tags weren't working correctly and sometimes was possible to see a tag multiple times in the database if it was connected to multiple Products.
- Created a new model `ProductTagCreate` which will be used to handle the create errors on Product Tags Batch Create endpoint.
- Updated the `ProductTagListMapper` to use `ProductTagCreate` when needed.
- `Product` now is able to encode tags.
- Created `ProductTagsViewController` which handle all the logic around Product Tags UI
- `ProductFormViewController` now is able to present and handle Product Tags

## Testing
1. Navigate to a product without tags, press "Add more details"  button, and choose tags.
2. Play with tags, trying to type them, and also try to add some tags from the suggestion list.
3. Add also some new tags (not already existing).
3. Press "Save", and make sure the tags are now updated based on the new ones.
3. Press "Update" in the Product Detail, and make sure that the tags remotely are updated correctly.

## GIF
<img src="https://user-images.githubusercontent.com/495617/88420653-82928000-cde7-11ea-9629-45b3cc5ddc1c.gif" width="300"/>


## Screenshot with dark mode enabled. 
(I'm not sure this is the final result that we want @Garance91540 with the dark mode enabled).
<img src="https://user-images.githubusercontent.com/495617/88420625-74dcfa80-cde7-11ea-9ebb-009b5831a80c.png" width=300 />


Update release notes:
